File: code/Java/strategy/duck/RubberDuck.java
package strategy.duck;

import strategy.fly.FlyNoWay;
import strategy.quack.MuteQuack;

public class RubberDuck extends Duck{

    public RubberDuck() {
        flyBehavior = new FlyNoWay();
        quackBehavior = new MuteQuack();
    }

    @Override
    public void display() {
        System.out.println("RubberDuck");
    }
}

File: code/C++/strategy/rubberduck.cpp
#include "rubberduck.h"

RubberDuck::RubberDuck() {
	this->flyBehavior = new FlyNoWay();
	this->quackBehavior = new MuteQuack();
}

void RubberDuck::dispaly() {
	std::cout << "RubberDuck" << std::endl;
}